File: tests/fixtures/sample.jl
struct Point
    x::Float64
    y::Float64
end

mutable struct Calculator
    value::Int
end

abstract type Shape end

module MathHelper
    function add(a, b)
        a + b
    end
end

function greet(name::String)
    "Hello, $name!"
end

function add(a::Int, b::Int)::Int
    a + b
end

magnitude(p::Point) = sqrt(p.x^2 + p.y^2)

File: tests/fixtures/sample.hs
module Sample where

data Point = Point { x :: Double, y :: Double }

newtype Age = Age Int

type Pair = (Int, Int)

class Renderable a where
    render :: a -> String

instance Renderable Point where
    render (Point x y) = "(" ++ show x ++ ", " ++ show y ++ ")"

add :: Int -> Int -> Int
add a b = a + b

greet :: String -> String
greet name = "Hello, " ++ name ++ "!"

magnitude :: Point -> Double
magnitude (Point x y) = sqrt (x * x + y * y)
